src/test/java/uk/gov/pay/directdebit/gatewayaccounts/resources/GatewayAccountResourceIT.java

```diff
@@ -284,18 +323,37 @@ public void shouldReturnBadRequestIfValidationFails() throws JsonProcessingExcep
                 .post(GATEWAY_ACCOUNTS_API_PATH)
                 .then()
                 .contentType(JSON)
-                .statusCode(Response.Status.BAD_REQUEST.getStatusCode())
-                .body("message", is("Field(s) missing: [service_name]"));
+                .statusCode(422)
```

Comment:
not sure why we return a 422 here and then 400 below, in near identical situation

Reply:
Its a good question. As some context the process as I understand it is that the JSON is first deserialised into an instance of `CreateGatewayAccountRequest` and then validation is performed to ensure all constraints are met. 

The `422` is out-of-the-box behaviour as part of the validation step. The request was successfully deserialised into an instance of `CreateGatewayAccountRequest` and then each value is validated, any constraint validations then returned with `422` since `indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions`.

The case where an invalid enum value is provided causes an exception that prevents deserialising into an instance of `CreateGatewayAccountRequest` and is thrown as part of the deserialisation and before the validation occurs. Again the out-of-the-box code used by Jackson for this scenario `400` which signals `that the server cannot or will not process the request due to something that is perceived to be a client error`.

The problem is that in both instances from a client's perspective the problem is an invalid value, not an invalid request format in the scenario below, however they receive slightly different error codes. It might be difficult to resolve this since being unable to deserialise into the target class results in a `JsonProcessingException` being thrown within the Jackson library, which we catch with the `JsonProcessingExceptionMapper` included herein. However, it wouldn't necessarily be appropriate to make that mapper return `422` because it would also catch the scenario when someone sends some nonsense for which `400` would be appropriate.  This could probably be resolved somehow...but I didn't spend too much time on it because its an internal api and they both signify and error with a meaningful message.